internal/cluster/conditions.go

```diff
@@ -19,3 +27,29 @@ func (c conditionId) String() string {
 func isVipDhcpAllocationSet(c *clusterPreprocessContext) bool {
 	return swag.BoolValue(c.cluster.VipDhcpAllocation)
 }
+
+func areAllHostsPreparedSuccessfully(c *clusterPreprocessContext) bool {
+	for _, h := range c.cluster.Hosts {
+		if swag.StringValue(h.Status) != models.HostStatusPreparingSuccessful {
+			return false
+		}
+	}
+	return true
+}
+
+func isInsufficientHostExists(c *clusterPreprocessContext) bool {
+	for _, h := range c.cluster.Hosts {
+		if swag.StringValue(h.Status) == models.HostStatusInsufficient {
+			return true
+		}
+	}
+	return false
+}
+
+func isClusterPreparationSucceeded(c *clusterPreprocessContext) bool {
+	return c.cluster.InstallationPreparationCompletionStatus == common.InstallationPreparationSucceeded
+}
+
+func isClusterPreparationFailed(c *clusterPreprocessContext) bool {
+	return c.cluster.InstallationPreparationCompletionStatus == common.InstallationPreparationFailed
```

Comment:
I see that you added `common.InstallationPreparationFailed` and `common.InstallationPreparationSucceeded` but no one is setting it

Reply:
Later commit.  This commit is not affecting behavior. That commit should will effect the behavior.  It should be the result of installation preparation instead of moving to another state.